src/net_processing.cpp

```diff
@@ -4364,6 +4364,12 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
 
     const auto current_time = GetTime<std::chrono::microseconds>();
 
+    if (pto->IsAddrFetchConn() && current_time - std::chrono::seconds(pto->nTimeConnected) > 10 * AVG_ADDRESS_BROADCAST_INTERVAL) {
```

Comment:
Why linking this timing to `AVG_ADDRESS_BROADCAST_INTERVAL`? It seems irrelevant.
Also, 5 minutes feels like a lot, if we thinking of a bootstrapping node hitting buggy/sybil nodes again and again.

Reply:
Ah yeah, I see the Poisson justification. Shouldn't it be linked to `AVG_LOCAL_ADDRESS_BROADCAST_INTERVAL` then? This is the relevant constant for the first Poisson delay, no?

Reply:
I don't think so: `AVG_LOCAL_ADDRESS_BROADCAST_INTERVAL=24h` only deals with our self-announcements, which are then added to `m_addrs_to_send`. The frequency with which we actually send out the content of `m_addrs_to_send` is controlled by  `AVG_ADDRESS_BROADCAST_INTERVAL`. 

I initially chose a timeout of 2min instead of 5min, but that would have lad to a premature disconnect in 1.8% of attempts, which seemed too high.
I think it makes sense to be a bit more conservative here, because we use AddrFetch only when we have specified the peer with `-seednode` or when we can't do DNS and thus connect manually to DNS seeds. We currently don't make AddrFetch connections to random nodes, and don't just choose another AddrFetch peer if we don't get an answer from the current one.

Reply:
ACK the second part justification.

Re first part: You are right, my confusion was because of 2 variables we have: `m_next_addr_send` and `m_next_local_addr_send`.